authorization: clarify/fix service accounts and workspace initialization (#996)
kcp-dev/kcp: docs/authorization.md

  name: workspace-admin
```

### Initializing Workspaces

> By default, workspaces are only accessible to a user if they are in phse `Ready`. Workspaces that are initializing

Reviewer: ```suggestion
By default, workspaces are only accessible to a user if they are in phase `Ready`. Workspaces that are initializing
```

> can be access only by users that are granted `initialize` verb on the `clusterworkspaces/content` resource in the 

Reviewer: ```suggestion
can be accessed only by users that are granted the `initialize` verb on the `clusterworkspaces/content` resource in the 
```

parent workspace.

> E.g. the workspace `root:org:ws:ws` initializing user is bound to a role of the following shape in `root:org:ws`:

Reviewer: ```suggestion
E.g. the actor that initializes the workspace `root:org:ws:ws` is bound to a role of the following shape in `root:org:ws`:
```

Note: groups added by the workspace content authorizer can be used for role bindings in that workspace.

It is possible to bind to roles and cluster roles in the bootstrap policy from a local policy `RoleBinding` or `ClusterRoleBinding`.

# Service Accounts

> Kubernetes service accounts is granted access to the workspaces they are defined in and that are ready. 

Reviewer: ```suggestion
Kubernetes service accounts are granted access to the workspaces they are defined when the workspace is in the `Ready` phase. 
```

> E.g. a service account "default" in `root:org:ws:ws` is granted access to `root:org:ws:ws`, and through the

Reviewer: Need to be more clear about the `:ws:ws` nesting here

Reviewer: What does "granted access" mean?

> workspace content authorizer it gains the `system:kcp:clusterworkspace:access` group membership.

Reviewer: Is this what it gains only during `Ready` phase?

> A service account can gain access (both pure access and `admin` access) to sub-workspaces defined in the workspace

Reviewer: How is the distinction between "pure" and "admin" important?
